jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -699,6 +699,31 @@ ecma_concat_ecma_strings (ecma_string_t *string1_p, /**< first ecma-string */
 } /* ecma_concat_ecma_strings */
 
 /**
+ * Join an array of ecma-strings and release them.
+ *
+ * @return a newly joined ecma-string
+ */
+ecma_string_t *
+ecma_join_ecma_strings (ecma_string_t **string_array, uint32_t n)
+{
+  JERRY_ASSERT (string_array != NULL);
+  JERRY_ASSERT (n > 0);
+
+  uint32_t i = 0;
+
+  for (; i < n - 1; ++i)
+  {
+    ecma_string_t *string1_p = string_array[i];
+    ecma_string_t *string2_p = string_array[i + 1];
+    string_array[i + 1] = ecma_concat_ecma_strings (string1_p, string2_p);
+    ecma_deref_ecma_string (string1_p);
+    ecma_deref_ecma_string (string2_p);
```

Comment:
You don't need to overwrite the string array, just keep the final string in a local variable.